engine-core/src/main/java/io/nosqlbench/engine/core/lifecycle/scenario/Scenario.java

```diff
@@ -278,6 +278,7 @@ private synchronized void runScenario() {
 
         Runtime.getRuntime().removeShutdownHook(scenarioShutdownHook);
         scenarioShutdownHook.run();
+        scenarioShutdownHook = null;
```

Comment:
I believe that the `.run()` above is essentially calling a synchronized `finish()` on this scenario instance from the scenarioShutdownHook.  If this is the case, what is the purpose of nulling out the scenarioShutdownHook here?

Reply:
This is a mutual exclusion check. The finish() method changes the volatile pointer which effectively signals whether or not the scenario was in control when it exited. The non-exceptional flow allows the Scenario it to deregister the hook and run it directly, but if this *doesn't* happen, then when the hook is run, an error is detected, as it was the side-effect of `Runtime.getRuntime().addShutdownHook(scenarioShutdownHook);` which triggered when the JVM exited. This is much like signal masking. It could probably be simplified, but the fix here was to avoid the present bug with minimal change.